### ISSUE TITLE
`azure storage account create` early validation for taken names

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/azure-xplat-cli#2120]
Small improvement tip: 

`azure storage account create` already prompts user for values, but it first prompts for "account name" which might be taken and then asks for 2 more questions, then if it's taken finally fails.

You got three options here:

1. Move `name` prompt to the end.
2. Validate availability of `name` just after prompting for it.
3. If user is already in prompt mode and account name is taken, just prompt again... Many commands have similar checks for empty parameters.

Example:

```
➜  ~  azure storage account create foo
info:    Executing command storage account create
+ Getting type
help:    Account Type:
  1) LRS
  2) ZRS
  3) GRS
  4) RAGRS
  5) PLRS
  : 1
+ Getting locations
help:    Location:
  1) Central US
  2) South Central US
  3) East US
  4) West US
  5) North Central US
  6) East US 2
  7) North Europe
  8) West Europe
  9) Southeast Asia
  10) East Asia
  11) Japan West
  12) Japan East
  13) Brazil South
  14) Australia Southeast
  15) Australia East
  16) Central India
  17) South India
  18) West India
  : 2
+ Creating storage account
error:   The storage account named 'foo' is already taken.
info:    Error information has been recorded to /Users/alp/.azure/azure.err
error:   storage account create command failed
```

**Comment by yaxia** [comment]
ETA of the fix: May release

**Comment by yaxia** [comment]
This is fixed with the release [0.10.0](https://github.com/Azure/azure-xplat-cli/tree/v0.10.0-May2016)
